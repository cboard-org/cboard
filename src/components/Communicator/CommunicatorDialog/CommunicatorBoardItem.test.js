import React from 'react';
import { shallowMatchSnapshot } from '../../../common/test_utils';
import CommunicatorBoardItem from './CommunicatorBoardItem.component';

jest.mock('./CommunicatorDialog.messages', () => {
  return {
    title: {
      id: 'cboard.components.CommunicatorDialog.title',
      defaultMessage: 'My Communicator'
    },
    menu: {
      id: 'cboard.components.CommunicatorDialog.menu',
      defaultMessage: 'Menu'
    },
    menuRootBoardOption: {
      id: 'cboard.components.CommunicatorDialog.menuRootBoardOption',
      defaultMessage: 'Set as Root Board'
    },
    menuPublishOption: {
      id: 'cboard.components.CommunicatorDialog.menuPublishOption',
      defaultMessage: 'Publish Board'
    },
    menuUnpublishOption: {
      id: 'cboard.components.CommunicatorDialog.menuUnpublishOption',
      defaultMessage: 'Unpublish Board'
    },
    emptyBoardsList: {
      id: 'cboard.components.CommunicatorDialog.emptyBoardsList',
      defaultMessage: 'No boards :('
    },
    loadNextPage: {
      id: 'cboard.components.CommunicatorDialog.loadNextPage',
      defaultMessage: 'Load More'
    },
    search: {
      id: 'cboard.components.CommunicatorDialog.search',
      defaultMessage: 'Search'
    },
    author: {
      id: 'cboard.components.CommunicatorDialog.author',
      defaultMessage: 'By {author}'
    },
    addBoard: {
      id: 'cboard.components.CommunicatorDialog.addBoard',
      defaultMessage: 'Add Board'
    },
    removeBoard: {
      id: 'cboard.components.CommunicatorDialog.removeBoard',
      defaultMessage: 'Remove Board'
    },
    communicatorBoards: {
      id: 'cboard.components.CommunicatorDialog.communicatorBoards',
      defaultMessage: 'Boards'
    },
    allBoards: {
      id: 'cboard.components.CommunicatorDialog.allBoards',
      defaultMessage: 'Public Boards'
    },
    myBoards: {
      id: 'cboard.components.CommunicatorDialog.myBoards',
      defaultMessage: 'All My Boards'
    },
    boardsQty: {
      id: 'cboard.components.CommunicatorDialog.boardsQty',
      defaultMessage: '{qty} boards'
    },
    helpAndSupport: {
      id: 'cboard.components.CommunicatorDialog.helpAndSupport',
      defaultMessage: 'Help And Support'
    },
    termsOfService: {
      id: 'cboard.components.CommunicatorDialog.termsOfService',
      defaultMessage: 'Terms Of Service'
    },
    tilesQty: {
      id: 'cboard.components.CommunicatorDialog.tilesQty',
      defaultMessage: '{qty} Tiles'
    },
    boardInfo: {
      id: 'cboard.components.CommunicatorDialog.boardInfo',
      defaultMessage: 'Board Information'
    },
    boardAddedToCommunicator: {
      id: 'cboard.components.CommunicatorDialog.boardAddedToCommunicator',
      defaultMessage: 'Board successfully added to your Communicator'
    },
    boardRemovedFromCommunicator: {
      id: 'cboard.components.CommunicatorDialog.boardRemovedFromCommunicator',
      defaultMessage: 'Board successfully removed from your Communicator'
    },
    close: {
      id: 'cboard.components.CommunicatorDialog.close',
      defaultMessage: 'Close'
    },
    boardInfoName: {
      id: 'cboard.components.CommunicatorDialog.boardInfoName',
      defaultMessage: 'Board name'
    },
    boardInfoAuthor: {
      id: 'cboard.components.CommunicatorDialog.boardInfoAuthor',
      defaultMessage: 'Board Author'
    },
    boardInfoTiles: {
      id: 'cboard.components.CommunicatorDialog.boardInfoTiles',
      defaultMessage: 'Number of tiles'
    },
    boardInfoId: {
      id: 'cboard.components.CommunicatorDialog.boardInfoId',
      defaultMessage: 'Board ID'
    }
  };
});

const intlMock = {
  formatMessage: ({ id }) => id
};

const COMPONENT_PROPS = {
  intl: intlMock,
  board: {
    id: 'someid',
    author: 'test author',
    nameKey: 'some.namekey.for.board',
    isPublic: false,
    tiles: []
  },
  addOrRemoveBoard: () => {},
  deleteBoard: () => {},
  publishBoardAction: () => {},
  setRootBoard: () => {}
};

describe('CommunicatorBoardItem tests', () => {
  test('default renderer', () => {
    shallowMatchSnapshot(<CommunicatorBoardItem {...COMPONENT_PROPS} />);
  });
});
