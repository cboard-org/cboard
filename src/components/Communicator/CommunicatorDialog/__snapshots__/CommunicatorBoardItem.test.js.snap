// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommunicatorBoardItem tests default renderer 1`] = `
<div
  className="CommunicatorDialog__boards__item"
>
  <div
    className="CommunicatorDialog__boards__item__image"
  >
    <div
      className="CommunicatorDialog__boards__item__image__empty"
    >
      <pure(ViewModuleIcon) />
    </div>
  </div>
  <div
    className="CommunicatorDialog__boards__item__data"
  >
    <div
      className="CommunicatorDialog__boards__item__data__title"
    >
      <WithStyles(ListItem)
        disableGutters={true}
      >
        <WithStyles(ListItemText)
          primary="someid"
          secondary="cboard.components.CommunicatorDialog.tilesQty"
        />
      </WithStyles(ListItem)>
    </div>
    <div
      className="CommunicatorDialog__boards__item__data__author"
    >
      cboard.components.CommunicatorDialog.author
    </div>
    <div
      className="CommunicatorDialog__boards__item__data__extra"
    />
  </div>
  <div
    className="CommunicatorDialog__boards__item__actions"
  />
</div>
`;
